### ISSUE TITLE
ai_gym list assignment index out of range

### ISSUE THREAD

**Issue by Chxinnn** [ultralytics/ultralytics#14064]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### YOLOv8 Component

Predict

### Bug

<img width="1116" alt="bug report" src="https://github.com/ultralytics/ultralytics/assets/93360224/7c4c9985-99c9-40b9-a219-64d1dbaeeb2d">

When I used the demo on  [the official website](https://docs.ultralytics.com/guides/workouts-monitoring/) to predict my video, the list overflowed. I found that the list created at the beginning was too small. In subsequent frames, the number of keypoints was greater than the capacity of the list.

<img width="1116" alt="bug report2" src="https://github.com/ultralytics/ultralytics/assets/93360224/b6bb8605-d208-48c2-ae7d-267c8fd73058">

### Environment

Ultralytics YOLOv8.2.45 🚀 Python-3.9.6 torch-2.3.1 CPU (Apple M3)
Setup complete ✅ (8 CPUs, 16.0 GB RAM, 156.1/460.4 GB disk)

OS                  macOS-14.3-arm64-arm-64bit
Environment         Darwin
Python              3.9.6
Install             pip
RAM                 16.00 GB
CPU                 Apple M3
CUDA                None

numpy               ✅ 1.26.4<2.0.0,>=1.23.5
matplotlib          ✅ 3.9.0>=3.3.0
opencv-python       ✅ 4.10.0.84>=4.6.0
pillow              ✅ 10.3.0>=7.1.2
pyyaml              ✅ 6.0.1>=5.3.1
requests            ✅ 2.32.3>=2.23.0
scipy               ✅ 1.13.1>=1.4.1
torch               ✅ 2.3.1>=1.8.0
torchvision         ✅ 0.18.1>=0.9.0
tqdm                ✅ 4.66.4>=4.64.0
psutil              ✅ 6.0.0
py-cpuinfo          ✅ 9.0.0
pandas              ✅ 2.2.2>=1.1.4
seaborn             ✅ 0.13.2>=0.11.0
ultralytics-thop    ✅ 2.0.0>=2.0.0

### Minimal Reproducible Example

```python
import cv2

from ultralytics import YOLO, solutions

model = YOLO("yolov8n-pose.pt")
cap = cv2.VideoCapture("fwc0.mp4")
assert cap.isOpened(), "Error reading video file"
w, h, fps = (int(cap.get(x)) for x in (cv2.CAP_PROP_FRAME_WIDTH, cv2.CAP_PROP_FRAME_HEIGHT, cv2.CAP_PROP_FPS))

video_writer = cv2.VideoWriter("workouts.avi", cv2.VideoWriter_fourcc(*"mp4v"), fps, (w, h))

gym_object = solutions.AIGym(
    line_thickness=2,
    view_img=True,
    pose_type="pushup",
    kpts_to_check=[6, 8, 10],
)

frame_count = 0
while cap.isOpened():
    success, im0 = cap.read()
    if not success:
        print("Video frame is empty or video processing has been successfully completed.")
        break
    frame_count += 1
    results = model.track(im0, verbose=False)  # Tracking recommended
    # results = model.predict(im0)  # Prediction also supported
    im0 = gym_object.start_counting(im0, results, frame_count)
    video_writer.write(im0)

cv2.destroyAllWindows()
video_writer.release()

```

### Are you willing to submit a PR?

- [X] Yes I'd like to help by submitting a PR!

**Comment by ambitious-octopus** [comment]
Hey @Chxinnn, thanks for reporting this issue. We've fixed it in PR #14074 . The error was caused by improper initialization of the lists. If there are no subjects to trace in the first frame, the lists weren't being initialized correctly.

**Comment by glenn-jocher** [comment]
@ambitious-octopus please ensure all PRs are linked to issue correctly. Mentioning the PR in a comment is unsufficient.

<img width="2000" alt="Screenshot 2024-06-29 at 17 58 53" src="https://github.com/ultralytics/ultralytics/assets/26833433/226436b4-1a66-4055-82c8-fce8499a9327">
